utils/test/oracle_report_helpers.py

```diff
@@ -0,0 +1,249 @@
+    assert hash_ == report.hex(), "HashConsensus points to unexpected report"
+    return members[0]
+
+
+def push_oracle_report(
+    *,
+    refSlot,
+    clBalance,
+    numValidators,
+    withdrawalVaultBalance,
+    elRewardsVaultBalance,
+    sharesRequestedToBurn,
+    simulatedShareRate,
+    stakingModuleIdsWithNewlyExitedValidators=[],
+    numExitedValidatorsByStakingModule=[],
+    withdrawalFinalizationBatches=[],
+    isBunkerMode=False,
+    extraDataFormat=0,
+    extraDataHash=ZERO_BYTES32,
+    extraDataItemsCount=0,
+    silent=False,
+):
+    if not silent:
+        print(f"Preparing oracle report for refSlot: {refSlot}")
+    consensusVersion = contracts.accounting_oracle.getConsensusVersion()
+    oracleVersion = contracts.accounting_oracle.getContractVersion()
+    (items, hash) = prepare_report(
+        refSlot=refSlot,
+        clBalance=clBalance,
+        numValidators=numValidators,
+        withdrawalVaultBalance=withdrawalVaultBalance,
+        elRewardsVaultBalance=elRewardsVaultBalance,
+        sharesRequestedToBurn=sharesRequestedToBurn,
+        simulatedShareRate=simulatedShareRate,
+        stakingModuleIdsWithNewlyExitedValidators=stakingModuleIdsWithNewlyExitedValidators,
+        numExitedValidatorsByStakingModule=numExitedValidatorsByStakingModule,
+        consensusVersion=consensusVersion,
+        withdrawalFinalizationBatches=withdrawalFinalizationBatches,
+        isBunkerMode=isBunkerMode,
+        extraDataFormat=extraDataFormat,
+        extraDataHash=extraDataHash,
+        extraDataItemsCount=extraDataItemsCount,
+    )
+    submitter = reach_consensus(refSlot, hash, consensusVersion, silent)
+    # print(contracts.oracle_report_sanity_checker.getOracleReportLimits())
+    report_tx = contracts.accounting_oracle.submitReportData(items, oracleVersion, {"from": submitter})
+    if not silent:
+        print(f"Submitted report data")
+    # TODO add support for extra data type 1
+    extra_report_tx = contracts.accounting_oracle.submitReportExtraDataEmpty({"from": submitter})
+    if not silent:
+        print(f"Submitted empty extra data report")
+
+    (
+        currentFrameRefSlot,
+        _,
+        mainDataHash,
+        mainDataSubmitted,
+        state_extraDataHash,
+        state_extraDataFormat,
+        extraDataSubmitted,
+        state_extraDataItemsCount,
+        extraDataItemsSubmitted,
+    ) = contracts.accounting_oracle.getProcessingState()
+
+    assert refSlot == currentFrameRefSlot
+    assert mainDataHash == hash.hex()
+    assert mainDataSubmitted
+    assert state_extraDataHash == extraDataHash.hex()
+    assert state_extraDataFormat == extraDataFormat
+    assert extraDataSubmitted
+    assert state_extraDataItemsCount == extraDataItemsCount
+    assert extraDataItemsSubmitted == extraDataItemsCount
+    return (report_tx, extra_report_tx)
+
+
+def simulate_report(*, refSlot, beaconValidators, postCLBalance, withdrawalVaultBalance, elRewardsVaultBalance):
```

Comment:
Would be great to allow `contracts.lido.handleOracleReport.call(...)` to be made for the provided historical block number: `contracts.lido.handleOracleReport.call(..., block_identifier=)`

The default option can be using the current block (at least for now)

Reply:
:brain: endgame: use the real ref_slot-related block by default